lib/engine-addon.js

```diff
@@ -387,10 +387,76 @@ module.exports = {
         return mergeTrees([
           externalTree,
           engineJSTree,
-          engineCSSTree
+          compiledEngineCSSTree
         ].filter(Boolean), { overwrite: true });
       };
 
+      this.compileLazyEngineStyles = function compileLazyEngineStyles(vendorTree, externalTree) {
+        var vendorCSSTree = buildVendorCSSTree.call(this, vendorTree);
+        var engineStylesOutputDir = 'engines-dist/' + this.name + '/assets/';
+
+        // Get base styles tree.
+        var engineStylesTree = this._treeFor('addon-styles');
```

Comment:
Its only called once per engine instance, the first invocation is for non-lazy engines and the second is for lazy engines.  I think we can definitely swap to memoize though, will update.